### ISSUE TITLE
How to load the custom font in pdfium.wasm

### ISSUE THREAD

**Issue by CCMKarthik123** [bblanchon/pdfium-binaries#163]
We retrieve the PDF image using pdfium.wasm on the client-side. However, in this scenario, custom fonts such as Turkish characters are not visible in the images. Please provide a solution to ensure that the correct fonts are retrieved from the pdfium.wasm file.

Please provide solution to set the custom font and get the correct image .

**Comment by CCMKarthik123** [comment]
Any solutions to resolve this issue?

**Comment by vecihibaltaci** [comment]
Is there any update on this?

**Comment by bblanchon** [comment]
@jerbob92, any idea how to fix this issue?

**Comment by jerbob92** [comment]
@KameshRajendran  This is not as per my suggestion.

`m_pUserFontPaths` should contain a path, not a filename.

You're also not putting the files itself in the virtual storage. So pdfium has no way to load your font.

> Note: This issue is affecting our production site, impacting a significant number of customers.

I don't get what you're trying to achieve with comments like this. It only makes me not want to help you anymore.

**Comment by KameshRajendran** [comment]
Hi @jerbob92 ,

I apologize if my previous comment came across as inappropriate or unhelpful. That was not my intention. I wanted to convey the urgency of the issue affecting our production site and a significant number of customers. I appreciate your assistance and understand the importance of clear and respectful communication.

Thank you for your understanding.

**Comment by Tamilselvan-Durairaj** [comment]
Hi @jerbob92 ,

We have configured the font file locations for our implementation and made modifications to the samples as described below:
![image](https://github.com/bblanchon/pdfium-binaries/assets/153176971/bc3c9627-8713-4716-8190-0f805746bc05)

The sample are committed in the below repository :
https://github.com/Tamilselvan-Durairaj/Pdfium

Publish Sample link for that:
https://tamilselvan-durairaj.github.io/Pdfium/

We encountered this issue and have been unable to find a proper solution. Could you please provide your guidance on how to resolve this?

**Comment by jerbob92** [comment]
@Tamilselvan-Durairaj thats a url, not a path, it has to be a path at which the font files are available in the virtual webassembly filesystem. Pdfium does not support loading fonts from a url.

**Comment by jerbob92** [comment]
@Tamilselvan-Durairaj are you sure you are actually using this project? Your source mentions https://github.com/paulocoutinhox/pdfium-lib

**Comment by jerbob92** [comment]
@bblanchon I have discovered that you can preload files in the compiled wasm.
If you build the wasm with the option: `--embed-file /usr/share/fonts@/usr/share/fonts` it will completely include the host system font path at that address. It makes the wasm file quite big (200MB, my fonts directory is quite big).

According to the pdfium source the base fonts are:

```
    "Courier",
    "Courier-Bold",
    "Courier-BoldOblique",
    "Courier-Oblique",
    "Helvetica",
    "Helvetica-Bold",
    "Helvetica-BoldOblique",
    "Helvetica-Oblique",
    "Times-Roman",
    "Times-Bold",
    "Times-BoldItalic",
    "Times-Italic",
    "Symbol",
    "ZapfDingbats",
```

We could also make a folder with just these fonts and then compile them into /usr/share/fonts in the wasm. I think the only question would be if this is actually allowed.

**Comment by Tamilselvan-Durairaj** [comment]
Hi @jerbob92 ,

Thank you for your support. Following your suggestion, we generated the file, and it has worked perfectly. Once again, We appreciate your valuable guidance in achieving this. 
I have one more question for you. Is it possible to load custom fonts into PDFium without embedding the fonts in the generated PDFium wasm file? The current approach has increased the file size of the wasm file.

Thanks You.

**Comment by jerbob92** [comment]
@Tamilselvan-Durairaj yes, the way to do is with the method that I have given before. Either by putting fonts in one of the predefined font folders using the virtual filesystem, or by giving a custom font folder with `m_pUserFontPaths` and then putting fonts in that font path using the virtual filesystem. You can ofcourse only compile the required font files into the wasm file instead of your whole fonts folder.

**Comment by inzanez** [comment]
@Tamilselvan-Durairaj 
how do you detect / know what font you are missing for the correct rendering?
